src/webServiceClient.ts

```diff
@@ -0,0 +1,138 @@
+import http = require('http');
+import https = require('https');
+import Transaction from './request/transaction';
+import * as models from './response/models';
+import { WebServiceClientError } from './types';
+
+interface ResponseError {
+  code?: string;
+  error?: string;
+}
+
+type servicePath = 'factors' | 'insights' | 'score';
+
+export default class WebServiceClient {
+  private accountID: string;
+  private host: string;
+  private licenseKey: string;
+  private timeout: number;
+
+  public constructor(accountID: string, licenseKey: string, timeout = 3000) {
+    this.accountID = accountID;
+    this.licenseKey = licenseKey;
+    this.timeout = timeout;
+    this.host = 'minfraud.maxmind.com';
+  }
+
+  public score(transaction: Transaction): Promise<models.Score> {
+    return this.responseFor<models.Score>(
+      'score',
+      transaction.toString(),
+      models.Score
+    );
+  }
+
+  private responseFor<T>(
+    path: servicePath,
+    postData: string,
+    modelClass: any
+  ): Promise<T> {
+    const parsedPath = `/minfraud/v2.0/${path}`;
+    const url = `https://${this.host}${parsedPath}`;
+
+    const options = {
+      auth: `${this.accountID}:${this.licenseKey}`,
+      headers: {
+        Accept: 'application/json',
+        'Content-Length': Buffer.byteLength(postData),
+        'Content-Type': 'application/json',
+        'User-Agent': `minfraud-api-node/${process.env.npm_package_version}`,
+      },
+      host: this.host,
+      method: 'POST',
+      path: parsedPath,
+      timeout: this.timeout,
+    };
+
+    return new Promise((resolve, reject) => {
+      const req = https.request(options, response => {
+        let data = '';
+
+        response.on('data', chunk => {
+          data += chunk;
+        });
+
+        response.on('end', () => {
+          try {
+            data = JSON.parse(data);
+          } catch {
+            return reject(this.handleError({}, response, url));
+          }
+
+          if (response.statusCode && response.statusCode !== 200) {
+            return reject(
+              this.handleError(data as ResponseError, response, url)
+            );
+          }
+
```

Comment:
Could we just "camelize" or transform the the whole response here, and not worry about "camelization" elsewhere in the app? Likewise, the snake casing transformation could occur prior to [line 58](https://github.com/maxmind/minfraud-api-node/pull/1/files#diff-febcda7767a9c31977b68edf094f1709R58).

Reply:
That could work, but I like how this function is focused on just server communication +error handling and the models handle all the data transformation work (casing, structure, etc).